go/arrow/compute/utils.go

```diff
@@ -81,3 +84,159 @@ func (b *bufferWriteSeeker) Seek(offset int64, whence int) (int64, error) {
 	b.pos = newpos
 	return int64(newpos), nil
 }
+
+func ensureDictionaryDecoded(vals ...arrow.DataType) {
```

Comment:
Can you add docstrings to all these helpers?

Reply:
Also, are there any tests for them?

Reply:
added